roles/network_plugin/calico/tasks/install.yml

```diff
@@ -174,25 +174,113 @@
         "logSeverityScreen": "Info",
         "nodeToNodeMeshEnabled": {{ nodeToNodeMeshEnabled|default('true') }} ,
         "asNumber": {{ global_as_num }} }} ' | {{ bin_dir }}/calicoctl.sh create --skip-exists -f -
-  run_once: true
-  delegate_to: "{{ groups['kube-master'][0] }}"
   changed_when: false
   when:
+    - inventory_hostname == groups['kube-master'][0]
     - calico_version is version('v3.0.0', '>=')
 
 - name: Calico | Set global as_num (legacy)
   command: "{{ bin_dir }}/calicoctl.sh config set asNumber {{ global_as_num }}"
-  run_once: true
   when:
+    - inventory_hostname == groups['kube-master'][0]
     - calico_version is version('v3.0.0', '<')
 
 - name: Calico | Disable node mesh (legacy)
   command: "{{ bin_dir }}/calicoctl.sh config set nodeToNodeMesh off"
-  run_once: yes
   when:
+    - inventory_hostname == groups['kube-master'][0]
     - calico_version is version('v3.0.0', '<')
     - nodeToMeshEnabled|default(True)
 
+- name: Calico | Configure peering with router(s) at global scope
+  shell: >
+   echo '{
+   "apiVersion": "projectcalico.org/v3",
+   "kind": "BGPPeer",
+   "metadata": {
+      "name": "global-{{ item.router_id }}"
+   },
+   "spec": {
+      "asNumber": "{{ item.as }}",
+      "peerIP": "{{ item.router_id }}"
+   }}' | {{ bin_dir }}/calicoctl.sh create --skip-exists -f -
+  retries: 4
+  delay: "{{ retry_stagger | random + 3 }}"
+  with_items:
+    - "{{ peers|selectattr('scope','defined')|selectattr('scope','equalto', 'global')|list|default([]) }}"
+  when:
+    - inventory_hostname == groups['kube-master'][0]
+    - calico_version | version_compare('v3.0.0', '>=')
```

Comment:
There is no need to check if Calico version is bigger than 3.0.0 since the other PR that removes 2.0 support will do a precheck for this

Reply:
This is a rewrite of the module, then the removing v2.6 support depends on this.. It's a much smaller patch